src/watch.ts

```diff
@@ -60,10 +60,10 @@ export class Watch {
             } else {
                 obj = JSON.parse(data) as WatchUpdate;
             }
-            if (obj.type && obj.object) {
+            if (typeof obj === 'object' && obj.object) {
                 callback(obj.type, obj.object);
             } else {
-                throw new Error(`unexpected object: ${JSON.stringify(obj)}`);
+                throw new Error(`unexpected ${typeof obj}: ${JSON.stringify(obj)}`);
```

Comment:
This will result in very compacted JSON; if you want to make the error message human-readable then consider `JSON.stringify(obj, undefined, 2)` (but this will potentially result in a multiline message, which you may feel is undesirable for other reasons!).

Reply:
Well, I feel like this should be handled in another issue: I just wanted to have a little clearer message.
The point is the exception was already unclear and multiline most of the times, we should review this in another way for sure but this is not the point of my pull request: making the code not crash during a documented event.

Reply:
We can absolutely have a better error management for this but I don't feel ready for a little refactor (first contribution here and low experience on Typescript)